GitVersionCore/GitPreparer.cs

```diff
@@ -202,10 +202,11 @@ private static Reference GetLocalReference(Repository repository, string branchN
             return repository.Refs.FirstOrDefault(localRef => string.Equals(localRef.CanonicalName, targetBranchName));
         }
 
-        private static DirectReference GetRemoteReference(Repository repository, string branchName, string repositoryUrl)
+        private static DirectReference GetRemoteReference(Repository repository, string branchName, Authentication authentication)
         {
             var targetBranchName = branchName.GetCanonicalBranchName();
-            var remoteReferences = repository.Network.ListReferences(repositoryUrl);
+            var remote = repository.Network.Remotes.Single();
```

Comment:
Will this method only be used on a CI server? Could it also be used on a developer environment? If that's the case, there are chances `Single()` will fail at runtime.